src/BlockTypes/ProductQuery.php

```diff
@@ -94,8 +94,12 @@ public function update_query( $pre_render, $parsed_block ) {
 	 */
 	public function update_rest_query( $args, $request ) {
 		$on_sale_query = $request->get_param( '__woocommerceOnSale' ) !== 'true' ? array() : $this->get_on_sale_products_query();
+		$stock_query   = empty( $request->get_param( '__woocommerceStockStatus' ) ) ? array() : $this->get_stock_status_query( $request->get_param( '__woocommerceStockStatus' ) );
 
-		return array_merge( $args, $on_sale_query );
+		return $this->merge_queries(
+			array_merge( $args, $on_sale_query ),
+			$stock_query
+		);
```

Comment:
What is the advantage of this over `array_merge( $args, $on_sale_query, $stock_query )`? We need to consider that we'll add more and more queries, already in https://github.com/woocommerce/woocommerce-blocks/pull/7687 I add another one which will conflict here, how should we do this in the future?

E.g.

```
return $this->merge_queries(
   array_merge( $args, $on_sale_query, $best_selling_query ),
   $stock_query
); 
``` 

How do we decide what's `$b` in the `merge_queries` call?

My suggestion here is to actually refactor the `merge_queries` call to accept multiple arguments, like most PHP array functions do. So the signature would be `merge_queries(array ...$queries)`. I had suggested it [here](https://github.com/woocommerce/woocommerce-blocks/commit/b8bf17efb32caf70c484e45d2a1e90a90c1bec90#r86518981) when this function was implemented, but we didn't go ahead with it because the advantages were not clear.

Now that we meet this problem, I think the advantages of having that is clearer, because we'll keep adding queries and we should just be able to do:

```
return $this->merge_queries(
   $args, $on_sale_query, $best_selling_query, $stock_query
); 
``` 

and don't worry about the internal implementation of that. What do you think?

Reply:
> What is the advantage of this over `array_merge( $args, $on_sale_query, $stock_query )`?

@sunyatasattva  With the current implementation of the class and structure of WP_Query, we have to use `merge_queries` for tax and meta query as `tax_query` and `meta_query` are arrays containing arrays.

> How do we decide what's `$b` in the `merge_queries` call?

As I mentioned above, queries containing `tax_query` and/or `meta_query` are `$b`.

> What do you think?

I strongly agree. `merge_queries` should be the source of trust in merging queries. I can refactor it either in this PR or a follow-up PR. Tagging @gigitux here for more perspectives :)

Reply:
I decided to go ahead and created https://github.com/woocommerce/woocommerce-blocks/pull/7697, so we have more things on the table to discuss.